packages/server/src/asyncQuestion/asyncQuestion.controller.ts

```diff
@@ -285,31 +288,54 @@ export class asyncQuestionController {
       );
     }
 
-    let updated = false;
-
     // Update allowed fields
     Object.keys(body).forEach((key) => {
       if (body[key] !== undefined && body[key] !== null) {
-        updated = true;
         question[key] = body[key];
       }
     });
 
     const updatedQuestion = await question.save();
 
-    // if a question is updated by student and is verified, mark all course users' entries as unread
-    if (updated && updatedQuestion.visible) {
-      await getRepository(UserCourseAsyncQuestionModel)
+    // Mark as new unread for all staff if the question needs attention
+    if (
+      body.status === asyncQuestionStatus.AIAnsweredNeedsAttention &&
+      oldQuestion.status !== asyncQuestionStatus.AIAnsweredNeedsAttention
+    ) {
+      await createQueryBuilder(UnreadAsyncQuestionModel)
```

Comment:
```suggestion
      await UnreadAsyncQuestionModel.createQueryBuilder()
```
current syntax (as well as getRepository()) is deprecated in later versions of typeorm